### PR TITLE
DataEng: Add test tasks and docker-compose for local deployment

### PR DESCRIPTION
**Pull request by KrasnovVitaliy**

### CODE REVIEW

File: dataeng/01-docker-comose/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+version: '3'
+services:
+  minio:
+    image: bitnami/minio:latest
```
Comment (IldarAlmakaev) [inline dataeng/01-docker-comose/docker-compose.yml:4]: Shouldn't we create a MinIO bucket as well?

File: dataeng/01-docker-comose/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+Docker-compose 1.26 or greater
+
+## Legend
+
+Docker compose with 3 applications (MINIO + postgres + pgadmin).
+
+### Instructions for running
+
+`docker-compose up -d db`
+
```
Comment (IldarAlmakaev) [inline dataeng/01-docker-comose/README.md:14]: What about stopping the docker-compose deployment?
`docker-compose down`?

File: dataeng/01-docker-comose/docker-compose.yml (hunk 1 above)
Comment (IldarAlmakaev) [reply dataeng/01-docker-comose/docker-compose.yml:4]: If we need it, I can do

File: dataeng/01-docker-comose/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
+## Prerequsite
+
```
Comment (IldarAlmakaev) [inline dataeng/01-docker-comose/README.md:2]: Prerequisites?

File: README.md [hunk 4]
```diff
@@ -18,6 +18,7 @@ Currently one specialization is available for internship. The Frontend recruitme
 
 - **[DevOps](./devops)**
 - **[Frontend](./frontend)**
+- **[DataEng](./dataeng)**
 
 Pick one and follow the directions in its README.md file.
 The test task consists of 2 parts: theory and practice.
```
Comment (IldarAlmakaev) [inline README.md:24]: I think we need to include overall info on what the docker-compose contains and  what they should do

File: dataeng/01-docker-comose/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    restart: always
+    environment:
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: postgres
+    volumes:
+      - ./postgres-data:/var/lib/postgresql
+    ports:
+      - 5432:5432
+
+  pgadmin:
```
Comment (alexeyzavyalov) [inline dataeng/01-docker-comose/docker-compose.yml:25]: maybe we can use this one https://github.com/dbcli/pgcli?

File: README.md (hunk 4 above)
Comment (KrasnovVitaliy) [reply README.md:24]: Added to task description

File: dataeng/01-docker-comose/docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    volumes:
+      - ./minio:/data
+    ports:
+      - 9000:9000
+
+  db:
+    image: postgres:13.3
+    restart: always
+    environment:
+      POSTGRES_USER: postgres
```
[the remark this answers is not in the document]
Comment (KrasnovVitaliy) [reply dataeng/01-docker-comose/docker-compose.yml:18]: done

File: dataeng/01-docker-comose/docker-compose.yml (hunk 5 above)
Comment (IldarAlmakaev) [reply dataeng/01-docker-comose/docker-compose.yml]: @alexeyzavyalov, I couldn't find a Docker image for that in Docker Hub

**Review by IldarAlmakaev — APPROVED** [review]

File: dataeng/README.md [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+```text
+first_name, last_name, birthts
+Ivan, Ivanov, 946674000000
+```
+
+### Data processing description
+
+1. Read csv file
+2. Try to find image for the same user.
```
Comment (IldarAlmakaev) [inline dataeng/README.md:33]: Может
_Match images for each user_

File: dataeng/README.md [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+3. Combine data from CSV and image path
+4. Update processed_data/output.csv CSV file and add new data. Important we can update data for previously processed
```
Comment (IldarAlmakaev) [inline dataeng/README.md:35]: _Combine the csv file data with image paths_?

File: dataeng/README.md [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+   user. In output CSV and DB we should not to duplicate records. Output CSV file format: user_id, first_name,
```
Comment (IldarAlmakaev) [inline dataeng/README.md:36]:
```suggestion
   user. In output CSV and DB we should not duplicate records. Output CSV file format: user_id, first_name,
```

File: dataeng/README.md [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+   last_name, birthts, img_path
+
+## Task
+
+Implement script to process files in src_data folder.
```
Comment (IldarAlmakaev) [inline dataeng/README.md:41]:
```suggestion
Implement a script to process files from the `src_data` folder.
```

File: dataeng/README.md [hunk 11]
```diff
@@ -0,0 +1,83 @@
+### Prerequisite
```
Comment (IldarAlmakaev) [inline dataeng/README.md:1]:
```suggestion
### Prerequisites
```

File: dataeng/README.md [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+**For example:**
+
+```text
+first_name, last_name, birthts
+Ivan, Ivanov, 946674000000
+```
+
+### Data processing description
+
+1. Read csv file
```
Comment (IldarAlmakaev) [inline dataeng/README.md:32]: А где находится этот csv файл?

File: dataeng/README.md [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+
+Implement script to process files in src_data folder.
+
+## Results delivery format
+
+Results should be implemented as a python script with demo data. Also should be
+provided the README.md file with the description of your solution.
+
+## Level 2
+The same as **Level 1** but with next differences.
```
Comment (IldarAlmakaev) [inline dataeng/README.md:49]:
```suggestion
The same as **Level 1** with the following extras.
```

File: dataeng/README.md [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+
+## Results delivery format
+
+Results should be implemented as a service. The service should periodically read source data and process it.
```
Comment (IldarAlmakaev) [inline dataeng/README.md:53]: Имеется в виду **web service**?

**Review by IldarAlmakaev — CHANGES_REQUESTED** [review]

**Review by IldarAlmakaev — APPROVED** [review]

**Pull request merged**